src/argus_htmx/incidents/views.py

```diff
@@ -55,119 +67,35 @@ def incident_row(request, pk: int):
-    return redirect("htmx:incident-detail", pk=pk)
-
-
-@require_POST
-def incident_detail_close(request, pk: int):
-    incident = get_object_or_404(Incident, id=pk)
-    if not incident.stateful:
-        LOG.warning(f"Attempt at closing the uncloseable {incident}")
-        messages.warning(request, f"Did not close {incident}, stateless incidents cannot be closed.")
-        return redirect("htmx:incident-detail", pk=pk)
-    form = DescriptionOptionalForm(request.POST or None)
-    if form.is_valid():
-        incident.set_closed(
-            request.user,
-            description=form.cleaned_data.get("description", ""),
-        )
-        LOG.info(f"{{ incident }} manually closed by {{ request.user }}")
-    return redirect("htmx:incident-detail", pk=pk)
-
-
-@require_POST
-def incident_detail_reopen(request, pk: int):
-    incident = get_object_or_404(Incident, id=pk)
-    if not incident.stateful:
-        LOG.warning(f"Attempt at reopening the unopenable {incident}")
-        messages.warning(request, f"Did not reopen {incident}, stateless incidents cannot be reopened.")
-        return redirect("htmx:incident-detail", pk=pk)
-    form = DescriptionOptionalForm(request.POST or None)
-    if form.is_valid():
-        incident.set_open(
-            request.user,
-            description=form.cleaned_data.get("description", ""),
-        )
-        LOG.info(f"{{ incident }} manually reopened by {{ request.user }}")
-    return redirect("htmx:incident-detail", pk=pk)
-
-
-@require_POST
-def incident_detail_add_ticket(request, pk: int):
-    incident = get_object_or_404(Incident, id=pk)
-    form = AddTicketUrlForm()
-    if request.POST:
-        form = AddTicketUrlForm(request.POST)
+    cleaned_form = None
+    if formdata:
+        form = formclass(formdata)
         if form.is_valid():
-            incident.ticket_url = form.cleaned_data["ticket_url"]
-            incident.save()
-
-    return redirect("htmx:incident-detail", pk=pk)
+            cleaned_form = form.cleaned_data
+    return cleaned_form
 
 
 @require_POST
-def incident_detail_edit_ticket(request, pk: int):
-    incident = get_object_or_404(Incident, id=pk)
-    form = EditTicketUrlForm()
-    if request.POST:
-        form = EditTicketUrlForm(request.POST)
-        if form.is_valid():
-            incident.ticket_url = form.cleaned_data["ticket_url"]
-            incident.save()
-
-    return redirect("htmx:incident-detail", pk=pk)
+def incidents_update(request: HtmxHttpRequest, pk: Optional[int] = None):
+    form_id = request.htmx.trigger_name or request.htmx.trigger
+    formclass, queryset = INCIDENT_UPDATE_FORM_NAMES.get(form_id, (None, None))
+    formdata = get_form_data(request, formclass)
+    if formdata:
+        # If no list of selected incident pks is provided, assume singular incident update
+        incident_ids = request.POST.getlist("selected_incidents")
+        if not incident_ids:
+            incident_ids = [get_object_or_404(Incident, id=pk).pk]
```

Comment:
You would get rid of this double if with two functions or always using formdata.